src/interpreter.rs

```diff
@@ -34,11 +43,19 @@ impl RuntimeError {
       | RuntimeError::InvalidType { location, .. }
       | RuntimeError::InvalidExpression { location, .. }
       | RuntimeError::IncorrectArgumentCount { location, .. } => *location,
+      RuntimeError::IndexOutOfBounds {
+        index,
+        length,
+        location,
+      } => *location,
+      RuntimeError::KeyNotFound { key, location } => *location,
+      RuntimeError::InvalidAssignment { location } => *location,
     }
   }
 }
 
 pub fn interpret(ast: Ast<'_>) -> RuntimeResult<()> {
+  // ast.print();
```

Comment:
It sounds like what you're using print for here can be done with the Debug trait: https://doc.rust-lang.org/std/fmt/trait.Debug.html

Essentially you can do
```rust
#[derive(Debug)]
struct MyStruct {
  hello: usize,
}
```

then do stuff like the following
```rust
let my_struct = MyStruct { hello: 123 };
dbg!(my_struct); // logs 'MyStruct { hello: 123 }'
println!("my struct is {:?}", my_struct); // logs 'my struct is MyStruct { hello: 123 }'
```